recipes/cad1/task2/baseline/evaluate.py

```diff
@@ -0,0 +1,327 @@
+"""Evaluate the enhanced signals using the HAAQI metric."""
+# pylint: disable=too-many-locals
+# pylint: disable=import-error
+
+import csv
+import hashlib
+import logging
+from pathlib import Path
+from typing import Tuple
+
+import hydra
+import numpy as np
+from omegaconf import DictConfig
+from scipy.io import wavfile
+from tqdm import tqdm
+
+from clarity.evaluator.haaqi import compute_haaqi
+from recipes.cad1.task2.baseline.audio_manager import AudioManager
+from recipes.cad1.task2.baseline.baseline_utils import (
+    load_listeners_and_scenes,
+    read_mp3,
+)
+from recipes.cad1.task2.baseline.car_scene_acoustics import CarSceneAcoustics
+
+logger = logging.getLogger(__name__)
+
+
+class ResultsFile:
+    """A utility class for writing results to a CSV file.
+
+    Attributes:
+        file_name (str): The name of the file to write results to.
+    """
+
+    def __init__(self, file_name):
+        """Initialize the ResultsFile instance.
+
+        Args:
+            file_name (str): The name of the file to write results to.
+        """
+        self.file_name = file_name
+
+    def write_header(self):
+        """Write the header row to the CSV file."""
+        with open(self.file_name, "w", encoding="utf-8") as csv_f:
+            csv_writer = csv.writer(
+                csv_f, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    "song",
+                    "listener",
+                    "score",
+                    "haaqi_left",
+                    "haaqi_right",
+                ]
+            )
+
+    def add_result(
+        self,
+        scene: str,
+        listener: str,
+        score: float,
+        haaqi_left: float,
+        haaqi_right: float,
+    ):
+        """Add a result to the CSV file.
+
+        Args:
+            scene (str): The name of the scene that the result is for.
+            listener (str): The name of the listener who submitted the result.
+            score (float): The combined score for the result.
+            haaqi_left (float): The HAAQI score for the left channel.
+            haaqi_right (float): The HAAQI score for the right channel.
+        """
+
+        logger.info(f"The combined score is {score}")
+
+        with open(self.file_name, "a", encoding="utf-8") as csv_f:
+            csv_writer = csv.writer(
+                csv_f, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    scene,
+                    listener,
+                    str(score),
+                    str(haaqi_left),
+                    str(haaqi_right),
+                ]
+            )
+
+
+def set_scene_seed(scene: str):
+    """Set a seed that is unique for the given song"""
```

Comment:
I like this approach of getting the seed from files.

Could be a little more descriptive here, not sure if I've understood it correctly.

```suggestion
    """Set a seed that is unique for the given song based onthe 'md5' `.hexdigest` of the scene itself."""
```